app/controllers/MovementsController.scala

```diff
@@ -111,31 +104,32 @@ class MovementsController @Inject()(
                     InternalServerError
                   }
                 }
+            case Left(error) =>
+              Logger.warn(s"Failed to create ArrivalMovement with the following error: $error")
+              Future.successful(BadRequest(s"Failed to create ArrivalMovement with the following error: $error"))
           }
       }
   }
 
   def putArrival(arrivalId: ArrivalId): Action[NodeSeq] = authenticateForWrite(arrivalId).async(parse.xml) {
     implicit request: ArrivalRequest[NodeSeq] =>
       arrivalMovementService
-        .messageAndMrn(request.arrival.nextMessageCorrelationId)(request.body)
-        .map {
-          case (message, mrn) =>
-            submitMessageService
-              .submitIe007Message(arrivalId, request.arrival.nextMessageId, message, mrn)
-              .map {
-                case SubmissionFailureInternal => InternalServerError
-                case SubmissionFailureExternal => BadGateway
-                case SubmissionFailureRejected => BadRequest("Failed schema validation")
-                case SubmissionSuccess =>
-                  Accepted("Message accepted")
-                    .withHeaders("Location" -> routes.MovementsController.getArrival(request.arrival.arrivalId).url)
-              }
-        }
-        .getOrElse {
-          Logger.warn("Invalid data: missing either DatOfPreMES9, TimOfPreMES10 or DocNumHEA5")
-          Future.successful(BadRequest("Invalid data: missing either DatOfPreMES9, TimOfPreMES10 or DocNumHEA5"))
-        }
+        .messageAndMrn(request.arrival.nextMessageCorrelationId)(request.body) match {
+        case Right((message, mrn)) =>
+          submitMessageService
+            .submitIe007Message(arrivalId, request.arrival.nextMessageId, message, mrn)
+            .map {
+              case SubmissionFailureInternal => InternalServerError
+              case SubmissionFailureExternal => BadGateway
+              case SubmissionFailureRejected => BadRequest("Failed schema validation")
+              case SubmissionSuccess =>
+                Accepted("Message accepted")
+                  .withHeaders("Location" -> routes.MovementsController.getArrival(request.arrival.arrivalId).url)
+            }
+        case Left(error) =>
+          Logger.warn(s"Failed to create message and MovementReferenceNumber with error: $error")
```

Comment:
I think it's worth properly showing the error. `Logger.error(msg, error)` or I am not sure if you can pass through an error object to the `Logger.warn` method, but we would need to, because in here we are throwing away the stack trace right now.